packages/wdio-browserstack-service/src/instrumentation/performance/performance-tester.ts

```diff
@@ -0,0 +1,329 @@
+        }, details), this.details[event] || {}))
+    }
+
+    static getProcessId() {
+        return `${process.pid}-${worker.threadId}`
+    }
+
+    static sleep = (ms = 100) => new Promise((resolve) => setTimeout(resolve, ms))
+
+    static async uploadEventsData() {
+        try {
+            let measures = []
+            if (fs.existsSync(this.jsonReportDirPath)) {
+                const files = (await fsPromise.readdir(this.jsonReportDirPath)).map(file => path.resolve(this.jsonReportDirPath, file))
+                measures = (await Promise.all(files.map((file) => fsPromise.readFile(file, 'utf-8')))).map(el => `[${el.slice(0, -1)}]`).map(el => JSON.parse(el)).flat()
+            }
+
+            if (this._measuredEvents.length > 0) {
+                measures = measures.concat(this._measuredEvents)
+            }
+
+            const date = new Date()
+            // yyyy-MM-dd'T'HH:mm:ss.SSSSSS Z
+            const options: Intl.DateTimeFormatOptions = {
+                timeZone: 'UTC',
+                year: 'numeric',
+                month: '2-digit',
+                day: '2-digit',
+                hour: '2-digit',
+                minute: '2-digit',
+                second: '2-digit',
+                fractionalSecondDigits: 3, // To include microseconds
+                hour12: false
+            }
+
+            // Format the date and replace the default separator for time zone
+            const formattedDate = new Intl.DateTimeFormat('en-GB', options)
+                .formatToParts(date)
+                .map(({ type, value }) => type === 'timeZoneName' ? 'Z' : value)
+                .join('')
+                .replace(',', 'T')
+
+            const payload = {
+                event_type: 'sdk_events',
+                data: {
+                    testhub_uuid: process.env.PERF_TESTHUB_UUID || process.env.PERF_SDK_RUN_ID,
+                    created_day: formattedDate,
+                    event_name: 'SDKFeaturePerformance',
+                    user_data: process.env.PERF_USER_NAME,
+                    host_info: JSON.stringify({
+                        hostname: hostname(),
+                        platform: platform(),
+                        type: type(),
+                        version: version(),
+                        arch: arch()
+                    }),
+                    event_json: { measures: measures }
+                }
+            }
+            const result = await got.post(`${EDS_URL}/send_sdk_events`, {
+                headers: {
+                    'content-type': 'application/json'
+                }, json: payload
+            })
+
+            BStackLogger.debug(`Successfully uploaded performance events ${util.format(result.body)}`)
+        } catch (er) {
+            BStackLogger.debug(`Failed to upload performance events ${util.format(er)}`)
+        }
+
+        try {
+            if (fs.existsSync(this.jsonReportDirPath)) {
```

Comment:
avoid mixing sync and async use of fs module, this function is async so you can use await everywhere
```suggestion
            if (await fsPromise.access(this.jsonReportDirPath).then(() => true, () => false)) {
```